test/src/components/shell/use-link-handler.test.tsx

```diff
@@ -0,0 +1,225 @@
+import React from 'react';
+import {describe, it, expect} from '@jest/globals';
+import {render, screen, fireEvent} from '@testing-library/preact';
+import userEvent from '@testing-library/user-event';
+import useLinkHandler, {
+    TrackedMouseEvent
+} from '~/components/shell/router-helpers/use-link-handler';
+import linkHelper from '~/helpers/link';
+import {useNavigate, MemoryRouter} from 'react-router-dom';
+
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    __esModule: true,
+    useNavigate: jest.fn()
+}));
+
+describe('use-link-handler', () => {
+    const user = userEvent.setup();
+    const notPrevented = jest.fn();
+    const Component = ({track = false}) => {
+        const linkHandler = useLinkHandler();
+        const onClick = React.useCallback(
+            (e: React.MouseEvent & {trackingInfo?: {[key:string]: string}}) => {
+                if (track) {
+                    (e).trackingInfo = {payload: 'junk'};
+                }
+                linkHandler(e as TrackedMouseEvent);
+                if (!e.defaultPrevented) {
+                    notPrevented(true);
+                }
+                e.preventDefault();
+            },
+            [linkHandler, track]
+        );
+
+        return <a href="https://dev.openstax.org/some-url" onClick={onClick} />;
+    };
+
+    jest.spyOn(window, 'open').mockImplementation(() => null);
+
+    it('short-circuits if not a valid URL click', async () => {
+        jest.spyOn(linkHelper, 'validUrlClick').mockReturnValue(false);
+
+        render(
+            <MemoryRouter initialEntries={['/']}>
+                <Component />
+            </MemoryRouter>
+        );
+        await user.click(screen.getByRole('link'));
+        expect(notPrevented).toBeCalledWith(true);
+        jest.resetAllMocks();
```

Comment:
`afterEach` on the parent describe block could work for all the reset calls, I think